corehq/apps/linked_domain/tests/test_view_helpers.py

```diff
@@ -0,0 +1,321 @@
+import uuid
+
+from django.test.testcases import TestCase
+
+from corehq.apps.app_manager.models import Application, LinkedApplication
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.fixtures.models import FixtureDataType, FixtureTypeField
+from corehq.apps.linked_domain.models import DomainLink
+from corehq.apps.linked_domain.view_helpers import (
+    build_app_view_model,
+    build_fixture_view_model,
+    build_keyword_view_model,
+    build_report_view_model,
+    get_apps,
+    get_fixtures,
+    get_keywords,
+    get_reports,
+)
+from corehq.apps.sms.models import Keyword
+from corehq.apps.userreports.dbaccessors import delete_all_report_configs
+from corehq.apps.userreports.models import (
+    DataSourceConfiguration,
+    ReportConfiguration,
+    ReportMeta,
+)
+
+
+def _create_report(domain, title="report", upstream_id=None, should_save=True):
```

Comment:
I'd have made these class-level methods, part of either a mixin or superclass that would be used by the other test classes, but I think that's just a stylistic difference, I can't come up with an especially compelling argument one way or the other.

Reply:
Ah interesting. I had mixed feelings putting those methods where they are. I was almost positive I was writing code that already existed, but am not sure if there is a realistic solution to that problem. I think my ideal is to house methods like these in a `test_util.py` at least for each app that the tests within that app reference. If adding to a mixin or base class is more conventional within HQ, I'm happy to do that.

Reply:
I can see a `test_util` working, HQ has a bunch of `test_util(s)` files. We do also have at least one or two text mixins - `TestXmlMixin` comes to mind. We also have some test utils that are context managers for things like creating a case for a single test. We don't have strong shared design principles across tests, so it varies a lot by django app.

No need to take action on this, I'm you'll be continuing to think about test design.